queue/index.js

```diff
@@ -0,0 +1,124 @@
+export class QueueChannel {
+  constructor(processing = false, data = []) {
+    this.processing = processing
+    this.data = data
+  }
+}
+
+export class Queue {
+  constructor(app, queues, key) {
+    this.app = app
+    this.queues = queues
+    this.key = key
+    this.clientId = undefined
+  }
+
+  setClientId(clientId) {
+    this.clientId = clientId
+    return this.clientId
+  }
+
+  add(action, meta) {
+    let queueKey
+    if (this.queues.size === 1) {
+      queueKey = 'main'
+    } else {
+      let labels = Array.from(this.queues.keys())
+      let type = action.type
+      let actionChannel = type.slice(0, type.indexOf('/'))
+      if (labels.includes(actionChannel)) {
+        queueKey = actionChannel
```

Comment:
We should not use channels as `queueKey`.

For instance, if we have `posts` and `comments` channels, their actions should not be processed in parallel. Otherwise, we can try to create a comment before comment’s post was created.